src/shims/lock.h

```diff
@@ -79,6 +79,28 @@ _dispatch_lock_owner(dispatch_lock lock_value)
 	return lock_value & DLOCK_OWNER_MASK;
 }
 
+#elif defined(__FreeBSD__)
```

Comment:
this is absolutely not functional, please do not add this until it is properly ported.